lte/gateway/python/magma/pipelined/app/classifier.py

```diff
@@ -0,0 +1,260 @@
+                   parser.OFPActionSetField(tun_ipv4_dst=enodeb_ip_addr)]
+        if sid:
+            actions.append(parser.OFPActionSetField(metadata=sid))
+
+        flows.add_flow(self._datapath, self.tbl_num, match, actions=actions,
+                       priority=priority, goto_table=self.next_table)
+
+        # Add flow for mtr port
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           in_port=self.config.mtr_port,
+                           ipv4_dst=ue_ip_adr)
+
+        actions = [parser.OFPActionSetField(tunnel_id=o_teid),
+                   parser.OFPActionSetField(tun_ipv4_dst=enodeb_ip_addr)]
+        if sid:
+            actions.append(parser.OFPActionSetField(metadata=sid))
+
+        flows.add_flow(self._datapath, self.tbl_num, match, actions=actions,
+                       priority=priority, goto_table=self.next_table)
+       
+        # Add ARP flow for LOCAL port
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_ARP,
+                           in_port=self._uplink_port, arp_tpa=ue_ip_adr)
+        if sid:
+            actions = [parser.OFPActionSetField(metadata=sid)]
+
+        flows.add_flow(self._datapath, self.tbl_num, match, actions=actions,
+                       priority=priority, goto_table=self.next_table)
+
+        # Add ARP flow for mtr port
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_ARP,
+                               in_port=self.config.mtr_port,
+                               arp_tpa=ue_ip_adr)
+        if sid:
+            actions = [parser.OFPActionSetField(metadata=sid)]
+
+        flows.add_flow(self._datapath, self.tbl_num, match, actions=actions,
+                       priority=priority, goto_table=self.next_table)
+
+
+    def _delete_tunnel_flows(self, i_teid:int, ue_ip_adr:str):
+
+        # Delete flow for gtp port
+        match = MagmaMatch(tunnel_id=i_teid, in_port=self.config.gtp_port)
+
+        flows.delete_flow(self._datapath, self.tbl_num, match)
+
+        # Delete flow for LOCAL port
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           in_port=self._uplink_port, ipv4_dst=ue_ip_adr)
+
+        flows.delete_flow(self._datapath, self.tbl_num, match)
+
+        # Delete flow for mtr port
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_IP,
+                           in_port=self.config.mtr_port,ipv4_dst=ue_ip_adr)
+
+        flows.delete_flow(self._datapath, self.tbl_num, match)
+
+        # Delete ARP flow for LOCAL port
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_ARP,
+                           in_port=self._uplink_port, arp_tpa=ue_ip_adr)
+
+        flows.delete_flow(self._datapath, self.tbl_num, match)
+
+        # Delete ARP flow for mtr port
+        match = MagmaMatch(eth_type=ether_types.ETH_TYPE_ARP,
+                           in_port=self.config.mtr_port, arp_tpa=ue_ip_adr)
+
+        flows.delete_flow(self._datapath, self.tbl_num, match)
+
```

Comment:
For flow delete/add cases, the arguments are different for each invoke like magma_match. So if we can add helper function for it then need to call same number times to that helper function. So it will not to avoid the calling the function to reduce.

Reply:
I was talking about following repeated pattern:
match = MagmaMatch(eth_type=ether_types.ETH_TYPE_ARP,
                           in_port=self.config.mtr_port, arp_tpa=ue_ip_adr)
flows.delete_flow(self._datapath, self.tbl_num, match)

But Thats fine for now.